core/tests/protractor-browserstack.conf.js

```diff
@@ -0,0 +1,243 @@
+  // If true, only chromedriver will be started, not a standalone selenium.
+  // Tests for browsers other than chrome will not run.
+  chromeOnly: false,
+  // Additional command line options to pass to selenium. For example,
+  // if you need to change the browser timeout, use
+  // seleniumArgs: ['-browserTimeout=60'],
+  seleniumArgs: [],
+
+  // If sauceUser and sauceKey are specified, seleniumServerJar will be ignored.
+  // The tests will be run remotely using SauceLabs.
+  sauceUser: null,
+  sauceKey: null,
+
+  // The address of a running selenium server. If specified, Protractor will
+  // connect to an already running instance of selenium. This usually looks like
+  // seleniumAddress: 'http://localhost:4444/wd/hub'
+  seleniumAddress: 'http://hub-cloud.browserstack.com/wd/hub',
+
+  // The timeout for each script run on the browser. This should be longer
+  // than the maximum time your application needs to stabilize between tasks.
+  // (Note that the hint tooltip has a 60-second timeout.)
+  allScriptsTimeout: 180000,
+
+  // ----- What tests to run -----
+  //
+  // When run without a command line parameter, all suites will run. If run
+  // with --suite=smoke, only the patterns matched by that suite will run.
+  suites: {
+    // When running the tests on browserstack, all suites should be added
+    // in 'full' individually. This is to parallelize the suites on
+    // browserstack.
+    full: [
+      'protractor/accessibility.js',
+      'protractor/learnerFlow.js',
+      'protractor/libraryFlow.js',
+      'protractor/loginFlow.js',
+      'protractor_mobile/navigation.js',
+      'protractor/subscriptionsFlow.js'
+    ],
+
+    accessibility: [
+      'protractor/accessibility.js'
+    ],
+
+    learner: [
+      'protractor/learnerFlow.js'
+    ],
+
+    library: [
+      'protractor/libraryFlow.js'
+    ],
+
+    login: [
+      'protractor/loginFlow.js'
+    ],
+
+    navigation: [
+      'protractor_mobile/navigation.js'
+    ],
+
+    subscriptions: [
+      'protractor/subscriptionsFlow.js'
+    ]
+  },
+
+  // ----- Capabilities to be passed to the webdriver instance ----
+  //
+  // For a full list of available capabilities, see
+  // https://code.google.com/p/selenium/wiki/DesiredCapabilities
+  // and
+  // https://code.google.com/p/selenium/source/browse/javascript/webdriver/capabilities.js
+  capabilities: {
+    'browserstack.user': process.env.USERNAME,
+    'browserstack.key': process.env.ACCESS_KEY,
+    'browserstack.local': true,
+    'browserstack.localIdentifier': process.env.BROWSERSTACK_LOCAL_IDENTIFIER,
+    'browserstack.debug': true,
+    'browserstack.networkLogs': true,
+    'browserstack.appium_version': '1.7.2',
+    device: process.env.DEVICE,
+    os_version: process.env.OS_VERSION,
+    real_mobile: 'true',
+    browserName: ''
+  },
+
+  // Code to start browserstack local before start of test
+  beforeLaunch: function() {
+    var checkSuites = function() {
```

Comment:
I don't think you need to check the number. Perhaps you could improve this to give a more useful error message: "Expected test file XXX to be in 'full', but it was not present." or "Test file XXX in 'full' does not have a corresponding individual suite."

Reply:
Done, thanks!